ci/build/npm-postinstall.sh

```diff
@@ -123,17 +123,48 @@ main() {
   fi
 }
 
-vscode_yarn() {
+install_with_yarn_or_npm() {
+  # NOTE@edvincent: We want to keep using the package manager that the end-user was using to install the package.
+  # This also ensures that when *we* run `yarn` in the development process, the yarn.lock file is used.
+  case "${npm_config_user_agent-}" in
+    yarn*)
+      if [ -f "yarn.lock" ]; then
+        if [[ $PWD/ = */lib/vscode/ ]]; then
+          yarn --production --frozen-lockfile --no-default-rc
```

Comment:
This is just replicating the current behavior introduced in https://github.com/coder/code-server/pull/5135/commits/7579dbbb06ba6c4b9a3e9ad94cdccfde15fcb5c7

@jsjoeio Remember why it's needed for `lib/vscode` but not for `lib/vscode/extensions`? Happy to add a comment to explain why this is needed, likely useful :D

Reply:
My memory is fuzzy so @jsjoeio can confirm but I think it was because of one or more of the following:

- The `.yarnrc` (which targets Electron) is in `lib/vscode` so it might not affect `lib/vscode/extensions` (idk if yarn travels upwards looking for a `.yarnrc` but it might).
- Even if the `.yarnrc` does affect `lib/vscode/extensions`, that directory has no native modules so it would have no influence anyway

That said, there is probably no harm in always including `--no-default-rc` for all our `yarn` calls.

Reply:
> That said, there is probably no harm in always including `--no-default-rc` for all our `yarn` calls.

Reverting to that then! (Which will fix my bad `bash` syntax that just caused a failed build too).

Reasoning makes sense. And yes, `yarn` does travel the parent directories looking for other `.yarnrc` files - and merges them - as per https://classic.yarnpkg.com/en/docs/yarnrc/

Reply:
Aha that makes sense, thanks for the link!

Reply:
I should have added a comment because my memory is fuzzy too 😅 Sounds like @code-asher nailed it though.